### DISCUSSION TITLE
How to use ApiContextInitializer and TelegramBotsApi()?

### DISCUSSION THREAD

**Discussion by proto-aiken-13** [rubenlagus/TelegramBots#1292] [Q&A]
Hello, I am developing a Telegram Bot using the guide given here:
https://monsterdeveloper.gitbook.io/java-telegram-bot-tutorial/chapter1

<img width="592" alt="image" src="https://github.com/rubenlagus/TelegramBots/assets/120178555/aacb3e58-d715-4ad9-9a7b-985982ddfe09">

One of the steps involves testing out our Telegram bot using a Main class. However, there are some problems:

The guide seems to use an older TelegramBot API (4.0.0), and I'm currently using a later build (6.8.0). From this build, it seems that many of the dependencies stipulated in the given guide are somewhat outdated (due to having different filepaths).

I can't seem to find the "ApiContextInitializer" directory. I assume this file has been refactored or removed during a release of the TelegramBots dependency? If so, may I ask what classes have replaced its role?

In addition, the guide uses an older version of TelegramBotsApi (where nothing is required for its constructor. But the new constructor seems to require a BotSession instance. If I want to, let's just say, get the bot to respond to my message, what class do I use/create?

Thanks in advance!

**Comment by rubenlagus** [comment]
Docs are now fully updated [here](https://rubenlagus.github.io/TelegramBotsDocumentation/java-telegram-bot-tutorial.html)
